src/firefly/js/visualize/ui/ImageTableRowViewer.jsx

```diff
@@ -48,8 +54,80 @@ export function ImageTableRowViewer({viewerId, makeRequestFromRow, defaultCutout
     useEffect(() => {
         if (!table || table.isFetching || !makeRequestFromRow) return;
         layoutImages(viewerId, Number(cutoutSize), Number(imageCnt), table, makeRequestFromRow);
+
+        //Change slide when table data, highlighted row, or image count was changed through UI
+        if (!slideChangeByArrows.current) { //don't change slide again when the highlighted row change was triggered by a slide change
+            const [visiblePlotsStartIdx,] = getVisiblePlotsRange(table.highlightedRow, table.totalRows, Number(imageCnt));
+            sliderRef.current?.slickGoTo(visiblePlotsStartIdx); //slider idx is always the 1st slide shown
+        }
+        else slideChangeByArrows.current = false;
     }, [imageCnt, table, cutoutSize]);
 
+    useEffect(()=>{
+        if (!activePlotId?.startsWith(plotIdRoot(viewerId))) return;
+
+        const activePlotRowNum = getPlotIdRowNum(viewerId, activePlotId);
+        if(table?.highlightedRow !== activePlotRowNum) {
+            //on active plot change, change highlighted row if not same
+            dispatchTableHighlight(tbl_id, activePlotRowNum);
+        }
+    }, [activePlotId]);
+
+    const sliderRef = useRef(null);
+    const slideChangeByArrows = useRef(false);
+
+    const makeImageSlider = (viewerItemIds, makeItemViewer) => {
```

Comment:
move makeImageSlider out of the component. It is just too big and make reading it hard.

Reply:
I moved it out in a component of its own